react-catalog/my-app/src/Catalog.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+import { readCatalog } from './read';
+import { type Product } from './data';
+import { Link } from 'react-router-dom';
+
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>();
+  useEffect(() => {
+    async function productCall() {
+      try {
+        const catalog = await readCatalog();
```

Comment:
You should display some type of "loading" message to the user when you make asynchronous calls. A spinner, an animating skeleton, a "loading..." message. Something, so that the user doesn't sit there staring at a blank page wondering if anything is really happening!

Reply:
I still suggest a loading indicator. It's optional here, but will be _required_ on your final project. You'd just as well learn how to do it now! Look into displaying a spinner over the page, for example.